packages/api/src/modules/providers/gispro/gispro.service.ts

```diff
@@ -0,0 +1,94 @@
+import GisproActionEntity from './entities/GisproActionEntity';
+import gisproRepository from './repositories/gispro.repository';
+import { isSiret, isAssociationName } from "../../../shared/Validators";
+import { DefaultObject, Siren, Siret } from '../../../@types';
+import GisproRequestAdapter from './adapters/GisproRequestAdapter';
+import DemandesSubventionsProvider from "../../demandes_subventions/interfaces/DemandesSubventionsProvider";
+import DemandeSubvention from '@api-subventions-asso/dto/search/DemandeSubventionDto';
+
+export const VALID_REQUEST_ERROR_CODE = {
+    INVALID_SIRET: 1,
+    INVALID_NAME: 2
+}
+
+export class GisproService implements DemandesSubventionsProvider {
+    public validEntity(entity: GisproActionEntity) {
+        if (!isSiret(entity.providerInformations.siret)) {
+            return { success: false, message: `INVALID SIRET FOR ${entity.providerInformations.siret}`, data: entity.providerInformations, code: VALID_REQUEST_ERROR_CODE.INVALID_SIRET };
+        }
+
+        if (!isAssociationName(entity.providerInformations.tier)) {
+            return { success: false, message: `INVALID NAME FOR ${entity.providerInformations.tier}`, data: entity.providerInformations, code: VALID_REQUEST_ERROR_CODE.INVALID_NAME };
+        }
+
+        return { success: true };
+    }
+
+    public async upsertMany(requests: GisproActionEntity[]) {
+        return await gisproRepository.upsertMany(requests);
+    }
+
+    public async insertMany(requests: GisproActionEntity[]) {
+        return await gisproRepository.insertMany(requests);
+    }
+
+    public async add(enitity: GisproActionEntity): Promise<{state: string, result: GisproActionEntity}> {
+        const existingFile = await gisproRepository.findByActionCode(enitity.providerInformations.codeAction);
+
+        if (existingFile) {
+            return {
+                state: "updated",
+                result: await gisproRepository.update(enitity),
+            };
+        }
+
+        return {
+            state: "created",
+            result: await gisproRepository.add(enitity),
+        };
+    }
+
+    public async findBySiret(siret: Siret) {
```

Comment:
idem ici, je pense qu'un MainService ou un ProviderService avec toutes les méthodes add(), findBySirent() qui utiliserait la référence à this.repository permettrait de faire tout ça sans redondance!

Reply:
Ça voudrait dire aussi faire un ProviderRepository ou au moins une interface que chaque repo d'un provider implémente. 

On à déjà des tickets pour ça non ?

Reply:
Yep je crois que ça peut matcher un des tickets refacto